### PR TITLE
apply new, more generic analysis format

### PR DESCRIPTION
**Pull request by kscieslinski**
Karton instances should be data-driven and so should not require any additional knowledge about from whom the task originated.
I've marked this or. as draft as I have tested it only locally.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,17 +17,39 @@ Scans analyses and samples with yara rules and spawns tasks with appropiate tags
...
+        "dumps_metadata": [
+            {"filename": <dump1_filename>, "base_address": <dump1_base_address>},
+            {"filename": <dump2_filename>, "base_address": <dump2_base_address>},
+            {"filename": <dump3_filename>, "base_address": <dump3_base_address>},
+            [...]
+        ],
+    }
+)
+```
+where `dumps_metadata` contains information about filename and base address for every memory dump in `dumps.zip`. The
```
Comment (nazywam) [inline README.md:44]:
```suggestion
where `dumps_metadata` contains information about filename and base address of every memory dump in `dumps.zip`. The
```

File: README.md [hunk 2]
```diff
@@ -17,17 +17,39 @@ Scans analyses and samples with yara rules and spawns tasks with appropiate tags
...
+following attributes are:
+- `filename` which is relative path to the dumps.zip contents;
+- `base_address` which hex-encoded base address for dump (leading `0x` is supported)
+You can specify multiple entries for the same file if the same memory dump was found on different base addresses.
+
+For the analysis type, yaramatcher runs rules against all dumps passed in `dumps.zip` resource and append result tags
```
Comment (nazywam) [inline README.md:50]: > all dumps passed in `dumps.zip`

I think this isn't 100% correct? It still iterates over the items in `dumps_metadata`

File: README.md [hunk 3]
```diff
@@ -17,17 +17,39 @@ Scans analyses and samples with yara rules and spawns tasks with appropiate tags
...
+            {"filename": <dump2_filename>, "base_address": <dump2_base_address>},
+            {"filename": <dump3_filename>, "base_address": <dump3_base_address>},
+            [...]
+        ],
+    }
+)
+```
+where `dumps_metadata` contains information about filename and base address for every memory dump in `dumps.zip`. The
+following attributes are:
+- `filename` which is relative path to the dumps.zip contents;
```
Comment (nazywam) [inline README.md:46]:
```suggestion
- `filename` - a relative path to dump in the dumps.zip archive;
```

File: README.md [hunk 4]
```diff
@@ -17,17 +17,39 @@ Scans analyses and samples with yara rules and spawns tasks with appropiate tags
...
+- `base_address` which hex-encoded base address for dump (leading `0x` is supported)
```
Comment (nazywam) [inline README.md:47]:
```suggestion
- `base_address` - hex-encoded dump base address (leading `0x` is supported)
```

File: README.md [hunk 5]
```diff
@@ -17,17 +17,39 @@ Scans analyses and samples with yara rules and spawns tasks with appropiate tags
...
-    "type": "analysis",
-    "kind": "drakrun"
-}, {
-    "type": "analysis",
-    "kind": "joesandbox"
+    "type": "analysis"
 }
 ```
 
+While `sample` type is self explanatory, the `analysis` type might be confusing. The `analysis` task is an output from
```
Comment (nazywam) [inline README.md:24]:
```suggestion
While `sample` type is self explanatory, the `analysis` type might be a bit more complicated. The `analysis` task is an output from
```

File: karton/yaramatcher/yaramatcher.py [hunk 6]
```diff
@@ -110,62 +107,20 @@ def scan_sample(self, sample: bytes) -> List[str]:
...
-            for rootdir, _dirs, files in os.walk(tmpdir):
-                for filename in files:
-                    with open(f"{rootdir}/{filename}", "rb") as dumpf:
-                        content = dumpf.read()
+        with dumps.extract_temporary() as tmpdir:  # type: ignore
+            for dump_metadata in dumps_metadata:
+                dump_path = (Path(tmpdir) / Path(dump_metadata["filename"])).resolve()
+                if not str(dump_path).startswith(tmpdir):
+                    continue
+                with open(dump_path, "rb") as dumpf:
```
Comment (nazywam) [inline karton/yaramatcher/yaramatcher.py:122]: If we're using pathlib and passing the content we might as well use `path.get_bytes()` here IMO

**Review by nazywam — APPROVED** [review]

File: karton/yaramatcher/yaramatcher.py (hunk 6 above)
Comment (kscieslinski) [reply karton/yaramatcher/yaramatcher.py]: Done

**Review by nazywam — APPROVED** [review]
:+1: I think we should @psrok1 approve as well and it'll be ready to go.

The release will probably deserve a major version bump :thinking:

**Pull request closed without merging**